Shopify/cli
# Add test event topic command

### WHY are these changes introduced?

This command will allow to request the system to trigger webhooks with sample content to a given destination

We initially added this new functionality in cli-plugins to wait for Editions announcements but, after ok from product and requirement of having the tool by default, we have moved the code from cli-plugins to cli 

### WHAT is this pull request doing?

Moves the code from cli-plugins to cli, to a command inside a topic named test, inside the main app topic
Removes the console option from the command

#### Extra considerations the reviewers should take

1. Meredith’s suggestions about naming (see https://github.com/Shopify/cli-plugins/pull/20#issuecomment-1293966739)
2. How the code is added to the cli project (folders structure, names, etc)
3. Any missing documentation 
4. Any considerations regarding the versioning and publishing (I don't know about that) 

:warning: ::warning: 🚧 
**After the first version of the PR and review from @pepicrft , we removed `port` and `url-path` and improved the documentation regarding the address parameter when you need a localhost delivery**
**This will also require changes in the documentation**

**Not decided on the final name yet**



### How to test your changes?

Use the command either against a remote destination (using https://webhook.site, for example) or against an app in localhost.
For the latter, create an app and use its local address as destination.
The app template validates the webhook origin using a shared secret configured in partners app.

```
yarn shopify app --help
Build Shopify apps

USAGE
  $ shopify app COMMAND

TOPICS
  app       Build Shopify apps
  auth      Auth operations
  event     Trigger delivery of a sample event topic payload to a designated address
...
```

```
yarn shopify event --help 
Trigger delivery of a sample event topic payload to a designated address

USAGE
  $ shopify event COMMAND

COMMANDS
  event trigger  Trigger delivery of a sample event topic payload to a designated address
```

```
yarn shopify event trigger --help
Trigger delivery of a sample event topic payload to a designated address

USAGE
  $ shopify event trigger [-h] [-t <value>] [-v <value>] [-m http|google-pub-sub|event-bridge] [-s <value>] [-a <value>]

FLAGS
  -a, --address=<value>           The URL where the webhook payload should be sent.
                                  For each delivery-method you will need a different address type:
                                  - http: Either https:// for remote or http://localhost:{port}/{url-path} for local
                                  - google-pub-sub: pubsub://{project-id}:{topic-id}
                                  - event-bridge: Amazon Resource Name (ARN) starting with arn:aws:events:
  -h, --help                      This help. When you run the trigger command the CLI will prompt you for any information that isn't passed using flags.
  -m, --delivery-method=<option>  Method chosen to deliver the topic payload. If not passed, it's inferred from the address
                                  <options: http|google-pub-sub|event-bridge>
  -s, --shared-secret=<value>     Shared secret used to build the X-Shopify-Hmac-SHA256 header.
  -t, --topic=<value>             The requested event topic.
  -v, --api-version=<value>       The API Version of the event topic.

DESCRIPTION
  Trigger delivery of a sample event topic payload to a designated address
```

### Post-release steps

The new command should be documented here: https://shopify.dev/apps/tools/cli/commands

### Measuring impact

How do we know this change was effective? Please choose one:


### Checklist
